pkg/fission-cli/cmd/environment/update.go

```diff
@@ -103,6 +103,9 @@ func updateExistingEnvironmentWithCmd(env *fv1.Environment, input cli.Input) (*f
 
 	if input.IsSet(flagkey.EnvPoolsize) {
 		env.Spec.Poolsize = input.Int(flagkey.EnvPoolsize)
+		if env.Spec.Poolsize == 0 {
```

Comment:
If the poolsize is set to zero - error should happen only in case of PoolManager. For NewDeploy it actually makes sense to keep poolsize as 0!